bothub-nlp-nlu-worker/bothub_nlp_nlu_worker/bothub_nlp_nlu/debug_parse.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+import json
+from collections import OrderedDict
+from rasa.nlu.test import remove_pretrained_extractors
+from lime.lime_text import LimeTextExplainer
+from .utils import update_interpreters
+from .utils import backend
+
+
+class DebugSentenceLime:
+    def __init__(self, interpreter, intention_names):
+        self.interpreter = interpreter
+        self.interpreter.pipeline = remove_pretrained_extractors(self.interpreter.pipeline)
+        self.intention_names = intention_names
+
+    def parse(self, text_list):
+        result_list = []
+        for text in text_list:
+            result_json = self.interpreter.parse(text)
+
+            idx_dict = {}  # fixing intent name to a index ex: {'violence': 0, 'immigration': 1, ... }
+            size = len(self.intention_names)
+            for i in range(size):
+                idx_dict[self.intention_names[i]] = i
+
+            intent_list = [0] * len(self.intention_names)
+            intent_name_list = [''] * len(self.intention_names)
+            size = len(result_json['intent_ranking'])
```

Comment:
use .get () instead of []
[https://stackoverflow.com/a/11041421/7768503](https://stackoverflow.com/a/11041421/7768503)